### PR TITLE
hard delete on bible audio

### PR DESCRIPTION
**Pull request by shimilgithub**
Issue #405

### CODE REVIEW

File: app/crud/contents_crud.py [hunk 1]
```diff
@@ -838,23 +856,24 @@ def get_bible_versification(db_, source_name):
         }
     return response
 
-def get_available_bible_books(db_, source_name,book_code=None, content_type=None,
-    biblecontent_id=None, **kwargs):
+def get_available_bible_books(db_, source_name,book_code=None, content_type=None,#pylint: disable = too-many-arguments,disable=too-many-locals
```
Comment (sijumoncy) [inline app/crud/contents_crud.py]: too-many-args pylint can be avoided by access bibleaudio_id via kwargs

File: app/crud/contents_crud.py [hunk 2]
```diff
@@ -838,23 +856,24 @@ def get_bible_versification(db_, source_name):
...
     if not source_name.endswith('_bible'):
         raise TypeException('The operation is supported only on bible')
     model_cls = db_models.dynamicTables[source_name]
-    # model_cls_audio = db_models.dynamicTables[source_name+"_audio"]
+    model_cls_audio = db_models.dynamicTables[source_name+"_audio"]
     query = db_.query(model_cls).options(joinedload(model_cls.book))
     fetched = None
     if biblecontent_id:
         query = query.filter(model_cls.bookContentId  == biblecontent_id)
+    if bibleaudio_id:
```
Comment (sijumoncy) [inline app/crud/contents_crud.py:876]: This is a common fucntion for getting bible content , based on content_type,
what happend if provide content_type other than audio and bibleaudio_id. 
Does this query returns proper response ?

File: app/crud/structurals_crud.py [hunk 3]
```diff
@@ -144,29 +144,33 @@ def restore_data(db_: Session, restored_item :schemas.RestoreIdentity):
...
     if db_restore.deletedFrom in content_class_map:
         model_cls = content_class_map[db_restore.deletedFrom]
     else:
         if not  get_sources(db_, table_name=db_restore.deletedFrom):
             raise NotAvailableException('Source not found in database')
-        source = get_sources(db_, table_name=db_restore.deletedFrom)[0]
-        source_name = source.sourceName
-        if source_name.endswith("bible") is False:
-            model_cls = db_models.dynamicTables[source.sourceName]
+        if db_restore.deletedFrom.endswith(("audio","cleaned")) is False:
```
Comment (sijumoncy) [inline app/crud/structurals_crud.py:159]: this means the control go to else part when the deleted from not endswith audio or cleaned. 
I saw the audio handled in else case . what about cleaned ?

File: app/schema/schema_content.py [hunk 4]
```diff
@@ -27,7 +27,7 @@ class Config:
 
 class AudioBible(BaseModel):
     '''Response object of Audio Bible'''
-    # audioId: int
+    audioId: int = None
```
Comment (sijumoncy) [inline app/schema/schema_content.py]: is this really need to make optional ? don't  we get unique id in every request

File: app/test/test_bibles.py [hunk 5]
```diff
@@ -1306,3 +1306,397 @@ def test_restoreitem_with_notavailable_source():
...
+    assert response.status_code == 201
+    assert len(response.json()['data']) == 6
+    assert response.json()['message'] == "Bible audios details uploaded successfully"
+    for item in response.json()['data']:
+        assert_positive_get_for_audio(item)
+    headers_auth = {"contentType": "application/json",#pylint: disable=redefined-outer-name
+                "accept": "application/json"}
+    headers_auth['Authorization'] = "Bearer"+" "+initial_test_users['VachanAdmin']['token']
+    #Check bible audio exist in dynamic table ater post
+    post_response = client.get(UNIT_URL+source_name+"/books?content_type=audio&udio_id"+str(audio_id),headers=headers_auth)
```
Comment (sijumoncy) [inline app/test/test_bibles.py]: How this test is passed ? 
content_type=audio&udio_id -  wrong param is passed here right ?

Comment (sijumoncy) [reply app/test/test_bibles.py]: is saw the response contain 6 items . The audio id is for book wise or single audio ?

File: app/test/test_bibles.py [hunk 6]
```diff
@@ -1306,3 +1306,397 @@ def test_restoreitem_with_notavailable_source():
...
+                    'Authorization': "Bearer"+" "+initial_test_users['VachanAdmin']['token']
+            }
+    response = client.delete(UNIT_URL+source_name+'/audios', headers=headers_va, json=data)
+    assert response.status_code == 200
+    assert response.json()['message'] ==\
+         f"Bible Audio with id {audio_id} deleted successfully"
+    delete_response = client.get(UNIT_URL+source_name+"/books?content_type=audio&audio_id"+str(audio_id),headers=headers_auth)
+    assert delete_response.status_code == 200
+    assert len(delete_response.json()) == 5
+    assert not delete_response.json()[0]['audio']['audioId'] == audio_id
```
Comment (sijumoncy) [inline app/test/test_bibles.py]: do you think , this need to test in a loop because 5 items are in response ?

File: app/test/test_bibles.py [hunk 7]
```diff
@@ -1306,3 +1306,397 @@ def test_restoreitem_with_notavailable_source():
...
+    assert response.json()['message'] == "Login Succesfull"
+    test_user_token = response.json()["token"]
+    headers_sa= {"contentType": "application/json",
+                    "accept": "application/json",
+                    'Authorization': "Bearer"+" "+test_user_token
+            }
+
+    data = {"sourceName":source_name}
+    response = client.delete(UNIT_URL, headers=headers_sa, json=data)
+    assert response.status_code == 404
```
Comment (sijumoncy) [inline app/test/test_bibles.py]: this need to be call an input validation error . 1 input item is missing ?

File: app/test/test_bibles.py [hunk 8]
```diff
@@ -1306,3 +1306,397 @@ def test_restoreitem_with_notavailable_source():
...
+    response = login(as_data)
+    assert response.json()['message'] == "Login Succesfull"
+    test_user_token = response.json()["token"]
+    headers_sa= {"contentType": "application/json",
+                    "accept": "application/json",
+                    'Authorization': "Bearer"+" "+test_user_token
+            }
+    data = {"itemId":audio_id}
+    response = client.delete(UNIT_URL, headers=headers_sa, json=data)
+    assert response.status_code == 404
```
Comment (sijumoncy) [inline app/test/test_bibles.py]: this also to be missing error ?

File: app/test/test_bibles.py [hunk 9]
```diff
@@ -1306,3 +1306,397 @@ def test_restoreitem_with_notavailable_source():
...
+    headers_auth = {"contentType": "application/json",#pylint: disable=redefined-outer-name
+                    "accept": "application/json",
+                    'Authorization': "Bearer"+" "+test_user_token
+            }
+    response = client.put(RESTORE_URL, headers=headers_auth, json=data)
+    assert response.status_code == 201
+    assert response.json()['message'] == \
+    f"Deleted Item with identity {deleteditem_id} restored successfully"
+    #Check bible book exists in dynamic table after restore
+    restore_response =  client.get(UNIT_URL+source_name+'/books?book_code=mat',\
```
Comment (sijumoncy) [inline app/test/test_bibles.py]: what is the content type here ? audio or all

File: app/test/test_bibles.py [hunk 10]
```diff
@@ -1306,3 +1306,397 @@ def test_restoreitem_with_notavailable_source():
...
+        headers=headers_auth)
+    assert restore_response.status_code == 200
+    assert len(restore_response.json()) == 1
+    for item in restore_response.json():
+        assert_positive_get_for_books(item)
+    #check item exists in cleaned table
```
Comment (sijumoncy) [inline app/test/test_bibles.py]: delete and restore audio does affect bible and cleaned ?

File: app/test/test_bibles.py [hunk 11]
```diff
@@ -1306,3 +1306,397 @@ def test_restoreitem_with_notavailable_source():
...
+            }
+    #Delete Associated Source
+    get_source_response = client.get(SOURCE_URL + "?source_name="+source_name, headers=headers_auth)
+    source_id = get_source_response.json()[0]["sourceId"]
+    data = {"itemId":source_id}
+    response = client.delete(SOURCE_URL, headers=headers_auth, json=data)
+    assert response.status_code == 200
+    #Restoring data
+    #Restore content with Super Admin after deleting source
+    restore_response = client.put(RESTORE_URL, headers=headers_auth, json=data)
```
Comment (sijumoncy) [inline app/test/test_bibles.py]: is this the same data you are passing here as the above one (delete source , restore audio )

**Review by sijumoncy — CHANGES_REQUESTED** [review]

File: app/crud/structurals_crud.py (hunk 3 above)
Comment (shimilgithub) [reply app/crud/structurals_crud.py:159]: > this means the control go to else part when the deleted from not endswith audio or cleaned. I saw the audio handled in else case . what about cleaned ?

On restore, cleaned table  has to be deleted along with bible books only.It should not be removed alone.Thats why that check is kept

File: app/test/test_bibles.py (hunk 10 above)
Comment (shimilgithub) [reply app/test/test_bibles.py]: No.bible and cleaned has the info about bible books.That test should not be in bile_audios.Removed and resolved

File: app/schema/schema_content.py (hunk 4 above)
Comment (shimilgithub) [reply app/schema/schema_content.py]: not necessary.removed "None" and resolved

Comment (sijumoncy) [reply app/schema/schema_content.py]: check does that make any issue in get bible books

File: app/test/test_bibles.py (hunk 7 above)
Comment (shimilgithub) [reply app/test/test_bibles.py]: Resolved

File: app/test/test_bibles.py (hunk 8 above)
Comment (shimilgithub) [reply app/test/test_bibles.py]: Resolved

File: app/test/test_bibles.py (hunk 11 above)
Comment (shimilgithub) [reply app/test/test_bibles.py]: Did you mean same as "test_restoreitem_with_notavailable_source" ?

Comment (sijumoncy) [reply app/test/test_bibles.py]: 1695 and 1699

Comment (shimilgithub) [reply app/test/test_bibles.py:1699]: > 

yes

Comment (shimilgithub) [reply app/test/test_bibles.py]: It should be different.resolved.

File: app/test/test_bibles.py (hunk 9 above)
Comment (shimilgithub) [reply app/test/test_bibles.py]: changed to audio

File: app/test/test_bibles.py (hunk 5 above)
Comment (shimilgithub) [reply app/test/test_bibles.py]: modified to fetch single item.both issues resolved

**Review by sijumoncy — APPROVED** [review]

**Pull request merged**